pkgs/applications/video/popcorntime/default.nix

```diff
@@ -0,0 +1,90 @@
+{ lib
+, stdenv
+, nodejs
+, mkYarnPackage
+, nwjs
+, fetchurl
+, fetchYarnDeps
+, fetchFromGitHub }:
+let
+
+  # We are using the Nwjs version referenced by popcorntime (0.44.5). Newer
+  # versions seem to fail.
+  # See:https://github.com/popcorn-official/popcorn-desktop/issues/2062
+  expectedNwjsVersion = "0.44.5";
+
+  nwjs' = nwjs.overrideAttrs (oldAttrs: rec {
+    version = expectedNwjsVersion;
+    src = fetchurl {
+      url = "https://dl.nwjs.io/v${version}/nwjs-v${version}-linux-x64.tar.gz";
+      sha256 = "sha256-LUs6wfQkpW8zH6WQOTcxSVOFGbQGtNg9EuiYGCsI3SQ=";
+    };
+  });
+
+in
+mkYarnPackage rec {
+  pname = "popcorntime";
+  version = "0.4.8";
+
+  src = fetchFromGitHub {
+    owner = "popcorn-official";
+    repo = "popcorn-desktop";
+    rev = "v${version}";
+    sha256 = "sha256-/mxfYGHvWM6VnZpH3T5YSkrtC07XW5MAETxwhBIlOb0=";
+  };
+
+  offlineCache = fetchYarnDeps {
+    inherit yarnLock;
+    sha256 = "sha256-wNzK7qdKNRxAiX3PYkzmGGCn4kZcWS9omNtyqoVljH4=";
+  };
+
+  packageJSON = ./package.json;
+  yarnLock = ./yarn.lock;
```

Comment:
if you needed to change the package json or the yarn.lock from upstream, I don't think you need to bring those files in nixpkgs.
If you needed to change something in the upstream files, it would be nice to have a comment about what it was so that the next person knowns what needs to be changed for an update.

Reply:
Oh good to know, I thought these files were mandatory ^^